integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/ServiceBusRestClient.java

```diff
@@ -101,36 +94,44 @@ public void sendMessage(byte[] message, String queuePath) {
 
         URI uri = convertToUri(resourceUri);
         log.debug("Calling {}", resourceUri);
-        ResponseEntity<String> response = restTemplate.exchange(uri, HttpMethod.POST, httpEntity, String.class);
-        if (!asList(HttpStatus.OK, HttpStatus.CREATED).contains(response.getStatusCode())) {
-            log.debug("{} got response {}, returning empty", resourceUri, response.getStatusCode().toString());
+        try {
+            ResponseEntity<String> response = restTemplate.exchange(uri, HttpMethod.POST, httpEntity, String.class);
+            if (!asList(HttpStatus.OK, HttpStatus.CREATED).contains(response.getStatusCode())) {
+                log.debug("{} got response {}, returning empty", resourceUri, response.getStatusCode().toString());
+                return Optional.empty();
+            }
+
+            ServiceBusMessage.ServiceBusMessageBuilder sbmBuilder = ServiceBusMessage.builder();
+            BrokerProperties brokerProperties = getBrokerProperties(response);
+
+            String messageId = brokerProperties.getMessageId();
+            sbmBuilder.lockToken(brokerProperties.getLockToken())
+                    .messageId(messageId)
+                    .sequenceNumber(brokerProperties.getSequenceNumber());
+
+            try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1141)